crates/router/src/workflows/revenue_recovery.rs

```diff
@@ -200,3 +214,159 @@ pub(crate) async fn get_schedule_time_to_retry_mit_payments(
+            BillingConnectorPaymentMethodDetails::Card(card_info) => card_info.card_issuer.clone(),
+            _ => None,
+        })
+        // will remove this after we are getting the card issuer from Stripe
+        .unwrap_or("CHASE".to_string());
+
+    let card_funding_opt = payment_intent
+        .feature_metadata
+        .as_ref()
+        .and_then(|revenue_recovery_data| {
+            revenue_recovery_data
+                .payment_revenue_recovery_metadata
+                .as_ref()
+        })
+        .map(|payment_metadata| payment_metadata.payment_method_subtype.to_string());
+
+    let card_funding_str = match card_funding_opt {
+        Some(cf_str) => cf_str,
+        None => {
+            return None;
+        }
+    };
+
+    let start_time_primitive = payment_intent.created_at;
+    // 1 hr after the actual start time
+    let modified_start_time_primitive =
+        start_time_primitive.saturating_add(time::Duration::hours(1));
+    let start_time_proto = date_time::convert_to_prost_timestamp(modified_start_time_primitive);
+
+    // Calculate end_time as start_time + 30 days
+    let end_time_primitive = start_time_primitive.saturating_add(time::Duration::days(30));
+    let end_time_proto = date_time::convert_to_prost_timestamp(end_time_primitive);
+
+    logger::debug!(
+        payment_intent_id = ?payment_intent.get_id(),
+        attempt_id = ?payment_attempt.get_id(),
+        message = "Card details for Recovery DeciderRequest",
+        first_error_message = %first_error_message,
+        billing_state = %billing_state,
+        card_funding = %card_funding_str,
+        card_network = %card_network_str,
+        card_issuer = %card_issuer_str
+    );
+
+    let decider_request = external_grpc_client::DeciderRequest {
+        first_error_message,
+        billing_state,
+        card_funding: card_funding_str,
+        card_network: card_network_str,
+        card_issuer: card_issuer_str,
+        start_time: Some(start_time_proto),
+        end_time: Some(end_time_proto),
+        retry_count: retry_count.into(),
+    };
+
+    // Clone the gRPC client into a mutable local variable, this is necessary because `decide_on_retry` requires `&mut self`,
+    // and we need an owned, mutable instance to call it.
+    let mut client = state.grpc_client.recovery_decider_client.clone();
+
+    match client
+        .decide_on_retry(decider_request, state.get_recovery_grpc_headers())
+        .await
+    {
+        Ok(grpc_response) => grpc_response
+            .retry_flag
+            .then_some(())
+            .and(grpc_response.retry_time)
+            .and_then(|prost_ts| {
+                match date_time::convert_from_prost_timestamp(&prost_ts) {
+                    Ok(pdt) => Some(pdt),
+                    Err(e) => {
+                        logger::error!(
+                            "Failed to convert retry_time from prost::Timestamp: {:?}",
+                            e
+                        );
```

Comment:
Nit: Clippy 1.88 would throw a warning.

```suggestion
                        logger::error!(
                            "Failed to convert retry_time from prost::Timestamp: {e:?}"
                        );
```